elife/php-dummies.sls

```diff
@@ -0,0 +1,72 @@
+# deploys orcid-dummy, hypothesis-dummy and similar
+# must have a web/index.php file
+
+{% for label, configuration in pillar.elife.php_dummies.items() %}
+{% set name = label | replace('_', '-') %} 
+{{ name }}-repository-reset: 
+    # to avoid
+    # stderr: fatal: could not set upstream of HEAD to origin/master when it does not point to any branch.
+    cmd.run:
+        - name: cd /srv/{{ name }} && git checkout master
+        - user: {{ pillar.elife.deploy_user.username }}
+        - onlyif:
+            - test -d /srv/{{ name }}
+
+{{ name }}-repository:
+    builder.git_latest:
+        - name: {{ configuration['repository'] }}
+        - identity: {{ pillar.elife.projects_builder.key or '' }}
+        - rev: master
+        - target: /srv/{{ name }}/
+        - force_fetch: True
+        - force_checkout: True
+        - force_reset: True
+        - require:
+            - {{ name }}-repository-reset
+
+    file.directory:
+        - name: /srv/{{ name }}
+        - user: {{ pillar.elife.deploy_user.username }}
+        - group: {{ pillar.elife.deploy_user.username }}
+        - recurse:
+            - user
+            - group
+        - require:
+            - builder: {{ name }}-repository
+
+    cmd.run:
+        - name: git checkout $(cat {{ configuration['pinned_revision_file'] }})
+        - cwd: /srv/{{ name }}
+        - user: {{ pillar.elife.deploy_user.username }}
+        - require:
+            - file: {{ name }}-repository
+
+{{ name }}-composer-install:
+    cmd.run:
+        {% if pillar.elife.env != 'dev' %}
+        - name: composer --no-interaction install --no-suggest --classmap-authoritative
+        {% else %}
+        - name: composer --no-interaction install --no-suggest
```

Comment:
Probably ok to separate `dev`, you might want to tweak the dummy at the same time as the main project.

Reply:
Least astonishment, ok